superdesk/content_types_async/service.py

```diff
@@ -0,0 +1,581 @@
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2025 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+from copy import deepcopy
+from typing import Any, Literal, cast
+
+import bson
+from bson import ObjectId
+from quart_babel import gettext as _
+
+from apps.auth import get_user_id
+from apps.desks import remove_profile_from_desks
+from apps.desks_async.desks_async_service import DesksAsyncService
```

Comment:
Rather than importing the service directly, which might cause cyclic imports in the end, you can import the ResourceModel and use that instead. i.e.

```python
from superdesk.types import DesksResourceModel

async def func_test():
    all_desks = await DesksResourceModel.get_service().get_all_list()
```